src/radical/pilot/bootstrapper/default_bootstrapper_anaconda.sh

```diff
@@ -0,0 +1,1459 @@
+# from the virtenv
+# NOTE: For some reasons, I have seen installations where 'scripts' go into
+#       bin/, and some where setuptools only changes them in place.  For now, 
+#       we allow for both -- but eventually (once the agent itself is small), 
+#       we may want to move it to bin ourself.  At that point, we probably
+#       have re-implemented pip... :/
+# FIXME: the second option should use $RP_MOD_PATH, or should derive the path
+#       from the imported rp modules __file__.
+if test "$RP_INSTALL_TARGET" = 'SANDBOX'
+then
+    PILOT_SCRIPT="$SANDBOX/rp_install/bin/radical-pilot-agent-$AGENT_TYPE.py"
+    if ! test -e "$PILOT_SCRIPT"
+    then
+        PILOT_SCRIPT="$SANDBOX/rp_install/lib/python$PYTHON_VERSION/site-packages/radical/pilot/agent/radical-pilot-agent-$AGENT_TYPE.py"
+    fi
+else
+    PILOT_SCRIPT="$VIRTENV/rp_install/bin/radical-pilot-agent-$AGENT_TYPE.py"
+    if ! test -e "$PILOT_SCRIPT"
+    then
+        PILOT_SCRIPT="$VIRTENV/rp_install/lib/python$PYTHON_VERSION/site-packages/radical/pilot/agent/radical-pilot-agent-$AGENT_TYPE.py"
+    fi
+fi
+
+
+# TODO: Can this be generalized with our new split-agent now?
+if test -z "$CCM"
+then
+    AGENT_CMD="$PYTHON $PILOT_SCRIPT"
+else
+    AGENT_CMD="ccmrun $PYTHON $AGENT_CMD"
+fi
+
+verify_rp_install
+
+echo
+echo "# -------------------------------------------------------------------"
+echo "# Launching radical-pilot-agent "
+echo "# CMDLINE: $AGENT_CMD"
+
+# At this point we expand the variables in $PREBOOTSTRAP2 to pick up the
+# changes made by the environment by pre_bootstrap_1.
+OLD_IFS=$IFS
+IFS=$'\n'
+for entry in $PREBOOTSTRAP2
+do
+    converted_entry=`eval echo $entry`
+    PREBOOTSTRAP2_EXPANDED="$PREBOOTSTRAP2_EXPANDED
+$converted_entry"
+done
+IFS=$OLD_IFS
+
+
+# we can't always lookup the ntp pool on compute nodes -- so do it once here,
+# and communicate the IP to the agent.  The agent may still not be able to
+# connect, but then a sensible timeout will kick in on ntplib.
+RADICAL_PILOT_NTPHOST=`dig +short 0.pool.ntp.org | grep -v -e ";;" -e "\.$" | head -n 1`
+
+# Before we start the (sub-)agent proper, we'll create a bootstrap_2.sh script
+# to do so.  For a single agent this is not needed -- but in the case where
+# we spawn out additional agent instances later, that script can be reused to
+# get proper # env settings etc, w/o running through bootstrap_1 again.
+# That includes pre_exec commands, virtualenv settings and sourcing (again),
+# and startup command).
+# We don't include any error checking right now, assuming that if the commands
+# worked once to get to this point, they should work again for the next agent.
+# Famous last words, I know...
+# Arguments to that script are passed on to the agent, which is specifically
+# done to distinguish agent instances.
+(cat <<EOT
+#!/bin/sh
+
+# some inspection for logging
+hostname
+
+# make sure we use the correct sandbox
+cd $SANDBOX
+
+# activate virtenv
+source activate $VIRTENV
```

Comment:
I don't get it.  `source` is a shell command which does not respect `$PATH` or anything else for its arguments.  So `activate` needs to be in the dir the `source` command is called in.  That seems to be `$SANDBOX` per 3 lines above.  How did the `activate` command get there?